diff --git a/conanfile.py b/conanfile.py
index d2a74aa..50d545a 100644
--- a/conanfile.py
+++ b/conanfile.py
@@ -55,9 +55,9 @@ USED_BOOST_COMPONENTS = [
 
 class QiPythonConan(ConanFile):
     requires = [
-        "boost/[~1.83]",
+        "boost/[~1.82]",
+        "openssl/[~3]",
         "pybind11/[^2.11]",
-        "qi/[~4]",
     ]
 
     test_requires = [
@@ -76,6 +76,12 @@ class QiPythonConan(ConanFile):
     }
 
     # Disable every components of Boost unless we actively use them.
+
+    def configure(self):
+        # Only set GNU-specific options if not using MSVC
+        if self.settings.get_safe("compiler") not in ["Visual Studio", "msvc"]:
+            self.options["compiler.libcxx"] = "libstdc++11"
+
     default_options.update(
         {
             f"boost/*:without_{_name}": (
