From da510f263fdd51e2fe3d23383a834f391afa9617 Mon Sep 17 00:00:00 2001
From: Simon Belgers <123083599+SBelgers@users.noreply.github.com>
Date: Wed, 18 Feb 2026 15:53:11 +0100
Subject: [PATCH 2/4] CMakeLists: use local libqi build and fix Boost

---
 CMakeLists.txt | 21 ++++++++++++++++-----
 1 file changed, 16 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c4dd4a0..0af5bc3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -78,10 +78,11 @@ find_package(
 ##############################################################################
 # Find Boost
 ##############################################################################
-find_package(
-  Boost REQUIRED
-  COMPONENTS thread
-)
+# find_package(
+#  Boost REQUIRED
+#  COMPONENTS thread
+#)
+# Provided by conan
 
 ##############################################################################
 # Find Pybind11
@@ -91,7 +92,12 @@ find_package(pybind11 REQUIRED)
 ##############################################################################
 # Find LibQi
 ##############################################################################
-find_package(qi REQUIRED)
+# find_package(qi REQUIRED)
+# Get the local build
+set(QI_LOCAL_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../libqi/build/release")
+if(NOT qi_DIR AND EXISTS "${QI_LOCAL_DIR}/qi-config.cmake")
+  set(qi_DIR "${QI_LOCAL_DIR}")
+endif()
 
 ##############################################################################
 # Convenience library: cxx_standard
@@ -285,6 +291,11 @@ install(
   DESTINATION qi
   COMPONENT Module
 )
+# Copy Python files to the build directory so the module is usable from there.
+foreach(file IN LISTS qi_python_py_files)
+  configure_file("${file}" "${file}" COPYONLY)
+endforeach()
+
 
 ##############################################################################
 # Tests
-- 
2.38.1.windows.1

